src/python/pants/rules/core/test.py

```diff
@@ -27,15 +27,24 @@ def fast_test(console, addresses):
   test_results = yield [Get(TestResult, Address, address.to_address()) for address in addresses]
   did_any_fail = False
   for address, test_result in zip(addresses, test_results):
+    if test_result.status == Status.FAILURE:
+      did_any_fail = True
     if test_result.stdout:
       console.write_stdout(
         "{} stdout:\n{}\n".format(
           address.reference(),
           console.red(test_result.stdout) if test_result.status == Status.FAILURE else test_result.stdout
         )
       )
-    if test_result.status == Status.FAILURE:
-      did_any_fail = True
+    if test_result.stderr:
+      # NB: we write to stdout, rather than to stderr, to avoid potential issues interleaving the
+      # two streams.
```

Comment:
Would be helpful to sanity check that this is the right behavior.

This decision came out of trying to implement `test_stderr`. I realized that if we wrote to `stderr`, we would still be writing some things to `stdout` like the test summary, and that it would be non-trivial to test how the two interleave.

Beyond testing, I think that the control flow of using the for loop means that for the end user, we would be interleaving correctly. If this is the case, perhaps out of correctness we should write to stderr and accept that it will make the tests a little bit more complex.

Reply:
Yep, writing to stdout here makes sense I think.